docs/plugins/best_practices.md

```diff
@@ -302,4 +302,75 @@ def change_style():
 get_settings().appearance.events.theme.connect(change_style)
 change_style()
 
-```
\ No newline at end of file
+```
+
+## Do not package your tests as top-level package
+
+
+```{note}
+If you are using the [napari plugin cookiecutter template](https://github.com/napari/cookiecutter-napari-plugin),
+your tests will be included within your package. That repository structure is not affected by this issue
+and does not need any further treatment.
+```
+
+Some repository structures place their testing logic next to the package, like:
+
+```
+my_package/
+  __init__.py
+  my_module.py
+tests/
+  conftest.py
+  test_my_module.py
+pyproject.toml
+README.md
+```
+
+Under these circumstances, your build backend (usually `setuptools`) might include `tests` as a
+separate package that will be installed next to `my_package`! 
+This is most of the time not wanted; e.g. do you want to do `import tests`? Probably not!
+Additionally, this unwanted behavior might cause installation issues with other projects.
+
+To avoid this problem, make sure you _exclude_ your top-level `tests` from the package:
```

Comment:
To be clear, the exclude part needs to be done **if** the directory structure is as above, correct?
Otherwise, one is encouraged to post tests in the subfolder, right like this:
```
 my_package/
   __init__.py
   my_module.py
   tests/
     conftest.py
     test_my_module.py
 pyproject.toml
 README.md
 ```